server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java

```diff
@@ -19,28 +19,76 @@
  */
 package org.sonar.process;
 
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.sonar.process.Lifecycle.State.HARD_STOPPING;
+import static org.sonar.process.Lifecycle.State.INIT;
+import static org.sonar.process.Lifecycle.State.RESTARTING;
+import static org.sonar.process.Lifecycle.State.STARTED;
+import static org.sonar.process.Lifecycle.State.STARTING;
+import static org.sonar.process.Lifecycle.State.STOPPED;
+import static org.sonar.process.Lifecycle.State.STOPPING;
+import static org.sonar.process.Lifecycle.State.values;
+
 public class Lifecycle {
+  private static final Logger LOG = LoggerFactory.getLogger(Lifecycle.class);
 
   public enum State {
-    INIT, STARTING, STARTED, STOPPING, STOPPED
+    INIT, STARTING, STARTED, RESTARTING, STOPPING, HARD_STOPPING, STOPPED
+  }
+
+  private static final Map<State, Set<State>> TRANSITIONS = buildTransitions();
+
+  private static Map<State, Set<State>> buildTransitions() {
+    Map<State, Set<State>> res = new HashMap<>(values().length);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Convert this Map to an EnumMap. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1640)